### DISCUSSION TITLE
Dremel Idea Builder 3D20 and other

### DISCUSSION THREAD

**Discussion by moonglow** [moonglow/FlashForge_Marlin#29] [Ideas]
That printers is actually FlashForge printers. As example we an compare motherboards ( they same as it used for FlashForge Dreamer ,but this is first splited revision and boards connected with FPC cable ). Dremel motherboard picture founded on Internet [Dremel MB](https://www.hubs.com/talk/uploads/db5060/original/3X/7/b/7bda526430e0750a3cf6856276831be6d5da241a.jpg). So we can easily run Marlin on it too ( or convert to FlashForge printer and vise versa ). But we have some cons what they firmware uses different encryption key. If someone can help with it, it be great! Need to dump original firmware bootloader with ST-Link or JLink programmer it will helps me to find encryption key.

**Comment by OneMonkeyArmy** [comment]
Since a long time, I know that The DREMEL 3D20 was a Flashforge Downgraded !!! and I was searching for a solution, not yesterday, this morning! I even talk with Flashforge and Dremel USA, to get some informations... but It was heading for a long shot!!... But boum... like magic.. you appear in my search..  AT first, I didn't believe my eyes. I downloaded the whole thing... And I installed the build version you provided, and it works!!  I have setup repetier, because you provided the IMPOSSIBLE TO GET usb driver for that board. I can't beleive the performances... I make a deal with you.... (if you agree) You gonna have to help me a bit on the software, and i am going to help you on the hardware... I already check the differences on the flashforge boards (main and driver) compare to the ones in dremel, and for a few dollars, i can add the missing components, and it will gives all the owner of 3d20, access to all the possibilities of dual extruder, heated bed... etc... Right now.. I am stuck.. i can't build the damn thing !!!  Just give me all the info on how to setup properly my folder that i open in VS code with platformio, and how to build, and you will have a TESTER working 20h per day !! And by the way, you deserve more than a few THANK YOU and a few coffee...  That I will take care of !!! (And if you have time to spend, I have a few  things to change in another project based on marlin) and this I will Gladly pay you with the currency of your choice!     NICE JOB !!

**Reply by moonglow** [reply]
@OneMonkeyArmy
Thank you for your kind words !)

>I am stuck.. i can't build the damn thing !!! Just give me all the info on how to setup properly my folder that i open in VS code with platformio, and how to build

Which OS you are using ? My main setup is: Ubuntu Linux ,but i think i can provide you info about Windows setup too, but i need to know OS first.

**Reply by OneMonkeyArmy** [reply]
Ok, I have compiled the tool for encription/decryption, and tested it, that works.. I have been able to compile the firmware, and encrypt it with the key you provided. And now the application provided by dremel to update the firmware no longer recognize the printer but the usb driver and the virtual comm works, because i can connect to the printer with repetier, pronterface even all my slicer softwares... that is the only problem left.. help me, i wont sleep until it works... !

**Reply by OneMonkeyArmy** [reply]
I forgat to tell you.. I have all the systems at home, Windows Mac OS and LINUX.. currently I am on a FU$%$%?% windows 10

**Reply by OneMonkeyArmy** [reply]
Finally it is as fast to just put the encrypted version on the micro sd card, and reboot the printer !  I will make an extension to connect in the slot and 30 cm of 8 wires and a new connector, on the side of the printer, this litte funny connector is hard to access.!!!  Case closed !!!  Now I have the full control of my 3D20... I will add a new support for the build plate and add an heated bed as well as a second extruder, and why not heated enclosure and cooling fan. I already added the missing parts and connectors on the coreBoard and followed the connection to the microcontroller so i know the exact pin to use. I already started to do the same thing on the driverboard. but i'm wondering where in the world you took all the informations needed to acheive what you did... humm !!! Because, neither Flashforge  or Dremel gives a lot of details... anyway, any suggestions or advice, I take everithing. Because I want to finish the Dremel ASAP, because there is a monster behind me waiting to be finished.. a 4 feet x 4 feet x  3 feet, Pellet extruder +  3d printer with 2 super volcano head 2.85mm, and a cnc router... the head on the gantry are changeable in less than 10 minutes, just 4 quick release, and reconnect the new connector, and boum, I have a different machine... you can see here some tests... https://www.youtube.com/watch?v=cU-LR5Mn9fY

**Reply by moonglow** [reply]
@OneMonkeyArmy 
>Case closed !!!

Wow... situation changes so fast :) That's nice what you already solved all questions ! But what was a difficult part ? Tutorial for Dremel firmware update/downgrade is not clearly enough shows needed steps?

>but i'm wondering where in the world you took all the informations needed to acheive what you did... humm !!!

First of all i was really unhappy about FF support team, they gives answers such "we are very happy what you ask us about this feature and many other users ask about it too, we sent request to our dev team.... " but nothing happing after that for years ... :) Some reverse engineering skills ( to recovery encryption algos, bootloader tricks and so on ), next i found on internet FF motherboard and bought it for cheap, it was my test setup, i use my tester and eyes to cover all pin connection on motherboard, i also idented most all IC on motherboard and next i start to add all this stuff to Marlin, write some drivers, change some existing code, make some config and voila it goes alive !  But Dremel machine has a little secret, they use different encryption key ( but as we already know, key looks like default key from FF team what they just not change at all ) and i start looking for Dremel owners with some engineering skills ( because i need to get bootloader dump ) and i found him [Tim Schoenmackers](https://github.com/timmehtimmeh), he helpme alot, provide me all needed data and do a test runs for Dremel firmware. So the main idea, that community around that project made that project alive.

>a 4 feet x 4 feet x 3 feet, Pellet extruder + 3d printer with 2 super volcano head 2.85mm, and a cnc router... the head on the gantry are changeable in less than 10 minutes

Real monster ! How you can combine so many skill together ? engineering + movie assistant ... that's looks very unusual

**Reply by OneMonkeyArmy** [reply]
I already tried to answer you by writting a 2 page email, but, I need a private way to communicate with you...  if I use e-mail, is that ok?
